lib/puppet/type/wls_datasource/connectionreservetimeoutseconds.rb

```diff
@@ -0,0 +1,12 @@
+newproperty(:connectionreservetimeoutseconds) do
+  include EasyType
+
+  desc 'The number of seconds after which a call to reserve a connection from the connection pool will timeout.'
+
+  defaultto '10'
```

Comment:
besides wls_setting, you can optional override it on the particular wls type, just like timeout